painttiming.bs

```diff
@@ -62,9 +89,28 @@ Formally, we consider the user agent to have "rendered" a document when it has p
 
     NOTE: The rendering pipeline is very complex, and the timestamp should be the latest timestamp the user agent is able to note in this pipeline (best effort). Typically the time at which the frame is submitted to the OS for display is recommended for this API.
 
+<dfn>Contentful</dfn>: An [=element=] or a [=pseudo-element=] counts as contentful when one or more of the following apply:
+* The element has a [=text node=] child, with at least one character, excluding [=document white space characters=].
+* The element is a [=replaced image=] representing a [=loaded=] image.
+* The element's [=computed=] style contains a [=loaded=] [=background-image=].
+* The element is a [=canvas=] with its [=context mode=] set to any value other than <code>none</code>.
+* The element is an [=svg element with rendered descendants=].
+* The element's [=computed=] style contains a [=contentful=] and [=paintable=] [=generated content pseudo-element=].
+
+<dfn>Paintable</dfn>: An [=element=] or a [=generated content pseudo-element=] counts as paintable when all of the following apply:
+* The element's [=computed=] [=visibility=] is <code>visible</code>.
+* The element and all of its ancestors have positive [=computed=] [=opacity=].
+
+    NOTE: there could be cases where a <code>paintable</code> element would not be visible to the user, for example in the case of text that has the same color as its background.
+    Those elements would still considered as paintable for the purpose of computing [=first contentful paint=].    
+
+* The element's bounding rect, as returned by [=getBoundingClientRect=] has positive right, bottom, width and height, after applying clipping as defined in [=Intersection rect algorithm=], but without clipping to the [=viewport=]. 
```

Comment:
It's still a bit unclear how the `Intersection rect algorithm` is being invoked, since it receives two parameters and only one is evident in here.

Reply:
Yea, I'll revise that. Also the intersection algo is very viewport-oriented.